packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -1293,7 +1295,17 @@ export function getBodyType(
       }
     }
     if (resourceType && bodyModel.name === "") {
-      bodyModel = resourceType;
+      const metadataInfo = createMetadataInfo(program);
+      const visibility = resolveRequestVisibility(
```

Comment:
Maybe some test cases to help understand the visibility impact?

Another point is if the type name would be duplicated if happening in different visibility scopes?

Reply:
yes, the visibility is not supported yet, so I am wondering if this could cause some issues or maybe it's better to hard code visibility?

Reply:
I don't understand what's the value to get a wrong model out here ?  @lirenhe  thoughts ?